Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -1342,14 +1345,26 @@ std::vector<double> DiscusMultipleScatteringCorrection::simulatePaths(
       std::transform(weights.begin(), weights.end(), sumOfWeights.begin(), sumOfWeights.begin(), std::plus<double>());
       std::transform(weights.begin(), weights.end(), countZeroWeights.begin(), countZeroWeights.begin(),
                      [](double d, int count) { return d > 0. ? count : count + 1; });
```

Comment:
Compiler also warns about precision loss here, I thought that might get caught by cppcheck?
```
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\include\algorithm(2184,36): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
1>C:\mantid-conda\mantid\Framework\Algorithms\src\DiscusMultipleScatteringCorrection.cpp(1347): message : see reference to function template instantiation '_OutIt std::transform<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>,Mantid::Algorithms::DiscusMultipleScatteringCorrection::simulatePaths::<lambda_971e91370384df0d12d827aa27767530>>(const _InIt1,const _InIt1,const _InIt2,_OutIt,_Fn)' being compiled
```

Reply:
I didn't see this compiler warning from Visual Studio but I can see why it might be a problem. I've fixed this by changing the type of the vector `countZeroWeights` to int